src/jdk.jartool/share/classes/sun/security/tools/jarsigner/Main.java

```diff
@@ -1128,6 +1133,170 @@ void verifyJar(String jarName)
         }
     }
 
+    private void crossCheckEntries(String jarName) throws Exception {
+        List<String> locEntries = new ArrayList<>();
```

Comment:
I think this can be a `HashSet` now that you don't care about the order when checking against the `cenEntries` later.

Reply:
Changed to `HashSet`.